Source/WebCore/svg/SVGPathElement.cpp

```diff
@@ -76,7 +77,9 @@ void SVGPathElement::attributeChanged(const QualifiedName& name, const AtomStrin
                 if (LIKELY(newDataSize <= (maxPathSegListCacheSize / 2))) {
                     pathSegListCacheSize += newDataSize;
                     while (pathSegListCacheSize > maxPathSegListCacheSize) {
+                        ASSERT(!cache.isEmpty());
                         auto iteratorToRemove = cache.random();
+                        ASSERT(iteratorToRemove != cache.end());
                         ASSERT(pathSegListCacheSize >= iteratorToRemove->value.size());
                         pathSegListCacheSize -= iteratorToRemove->value.size();
                         cache.remove(iteratorToRemove);
```

Comment:
I think the change in `SVGPathElement::attributeChanged()` is getting bigger and it became harder to read. I think this can be solved by moving the caching to a separate class say `PathSegListCache`. This class should manage adding values to the cache and retrieving existing values from the cache. This will allow early returns which makes the code more readable. The above code may be simplified to be like this:
```
    if (newValue.isEmpty())
        m_pathSegList->baseVal()->updateByteStreamData({ });
    else if (auto* data = PathSegListCache::singleton().get(newValue))
        m_pathSegList->baseVal()->updateByteStreamData(*data);
    else if (m_pathSegList->baseVal()->parse(newValue))
        PathSegListCache::singleton().add(newValue, m_pathSegList->baseVal()->existingPathByteStream().data())
    else
        document().accessSVGExtensions().reportError("Problem parsing d=\"" + newValue + "\"");
```